topsites.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/python
+import sys
+import time
+import datetime
+import hmac
+import hashlib
+import base64
+import collections
+import requests
+
+from urllib import urlencode
+from xml.dom.minidom import parseString
+
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, Integer, String
+from sqlalchemy.orm import sessionmaker
+
+# Constant for alexa top site API
+ATS_ACTION_NAME = "TopSites"
+ATS_RESPONSE_GROUP_NAME = "Country"
+ATS_SERVICE_HOST = "ats.amazonaws.com"
+ATS_AWS_BASE_URL = "http://" + ATS_SERVICE_HOST + "/?"
+
+ATS_DATEFORMAT = "%Y-%m-%dT%H:%M:%S.{}Z"
+ATS_HASH_ALGORITHM = "HmacSHA256"
+
+# Regions/Countries to dump to topsites.db
+regions = {
+    "": (10000, 1),  # Global
+    "US": (1000, 2),
+    "FR": (1000, 2),
+    "IN": (1000, 2),
+    "DE": (1000, 2),
+    "TW": (1000, 2),
+    "ID": (1000, 2),
+    "HK": (1000, 2),
+    "SG": (1000, 2),
+    "PL": (1000, 2),
+    "GB": (1000, 2),
+    "RU": (1000, 2),
+}
+
+# Cache parsed sites, change priority if raised
+topsites = {}
+
+engine = create_engine('sqlite:///topsites.db', echo=True)
+Base = declarative_base()
+Session = sessionmaker(bind=engine)
+session = Session()
+
+
+# sqlalchemy base object for a top site
+class Site(Base):
+    __tablename__ = "topsites"
+
+    url = Column(String, primary_key=True)
+    priority = Column(Integer)
+    country_code = Column(String)
+    ranking = Column(Integer)
+
+    def __init__(self, url, priority, country_code, ranking):
+        self.url = url
+        self.priority = priority
+        self.country_code = country_code
+        self.ranking = ranking
+
+    def __repr__(self):
+        return "<Site('%d','%d' '%s', '%s')>" % (self.priority, self.ranking, self.url, self.country_code)
+
```

Comment:
I'll remove __repr__ since it's just for my development and debugging, not used in real world now.